lib/constructs/interface.js

```diff
@@ -22,8 +52,20 @@ class Interface {
     this.mixins = [];
   }
 
+  get supportsIndexedProperties() {
```

Comment:
Since these are relatively expensive operations (O(n) in number of members) I think they should not be getters. Just `iface.supportsIndexedProperties()` makes sense to me.